apps/src/templates/sectionsRefresh/CoteacherSettings.jsx

```diff
@@ -0,0 +1,122 @@
+import PropTypes from 'prop-types';
+import React, {useMemo, useState} from 'react';
+import i18n from '@cdo/locale';
+
+import styles from './coteacher-settings.module.scss';
+import Button from '../Button';
+import {Figcaption} from '@cdo/apps/componentLibrary/typography';
+import FontAwesome from '../FontAwesome';
+import classNames from 'classnames';
+
+export default function CoteacherSettings({
+  sectionInstructors,
+  addCoteacher,
+  coteachersToAdd,
+}) {
+  const [inputValue, setInputValue] = useState('');
+  const [addError, setAddError] = useState('');
+
+  const handleInputChange = event => {
+    setInputValue(event.target.value);
+  };
+
+  const handleAddEmail = e => {
+    e.preventDefault();
+    const newEmail = inputValue;
+    console.log(sectionInstructors);
+    if (newEmail === '') {
+      setAddError(i18n.coteacherAddNoEmail());
+      return;
+    }
+    if (!newEmail.includes('@')) {
+      setAddError(i18n.coteacherAddInvalidEmail({email: newEmail}));
+      return;
+    }
+    if (
+      sectionInstructors.some(
+        instructor => instructor.instructorEmail === newEmail
+      ) ||
+      coteachersToAdd.some(email => email === newEmail)
+    ) {
+      setAddError(i18n.coteacherAddAlreadyExists({email: newEmail}));
+      return;
+    }
+
+    addCoteacher(newEmail);
+    setAddError('');
+    setInputValue('');
+  };
+
+  // coteacher count is teachers to add + the existing teachers - the primary teacher.
+  const numCoteachers = useMemo(
+    () => sectionInstructors.length + coteachersToAdd.length - 1,
```

Comment:
1) Is there any case where either `sectionInstructors` or `coteachersToAdd` might be null or otherwise not-arrays, making this calculation problematic?
2) Will this be off by 1 when creating a new section and there isn't a `sectionInstructors` entry for the main instructor yet?

Reply:
1. `coteachersToAdd` will never be null, but I did need to add some checking for `sectionInstructors`
2. fixed